integration_tests/src/janus.rs

```diff
@@ -56,3 +85,191 @@ impl<'a> Janus<'a> {
             .get_host_port_ipv4(Aggregator::INTERNAL_SERVING_PORT)
     }
 }
+
+/// Represents a running Janus test instance in this process.
+pub struct JanusInProcess {
```

Comment:
There's loads of other tests (such as in `janus_aggregator::aggregator::http_handlers`) where we set up an in-process instance of Janus by calling `janus_aggregator::aggregator::http_handlers::aggregator_handler`. I wonder if we should adapt those to use this entry point, or otherwise unify them? Though I don't think that needs to happen in this PR.

Reply:
Using the handler directly in those tests means they can run without binding to a port, skip both sides of the HTTP protocol, and manipulate requests and responses in memory. (using `trillium-testing`) I think it makes sense to keep things as they are, because they each have a different area of interest.